### ISSUE TITLE
[Cache] Don't store hop-by-hop headers

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/tempesta#409]
Keep-Alive and Server are hop-by-hop headers and should not be cached. `hbh_hdrs` must be adjusted and the headers must be made special to be analyzed quickly in `tfw_cache_copy_resp()`. 

Relates to #407.

**Comment by krizhanovsky** [comment]
See also [What Proxies Must Do](https://www.mnot.net/blog/2011/07/11/what_proxies_must_do) and particularly the referenced [spec](http://tools.ietf.org/html/draft-ietf-httpbis-p1-messaging-14#section-7.1.3)

**Comment by ikoveshnikov** [comment]
Actual problem differs from mentioned above: hop-by-hop headers are forwarded as end-to-end headers

In [spec mentioned above](https://tools.ietf.org/html/draft-ietf-httpbis-p1-messaging-14#section-7.1.3) 8 headers are marked as a hop-by-hop. But the [final document RFC7230](https://tools.ietf.org/html/rfc7230#section-6.1) is more strict than a draft and close to [article What Proxies Must Do](https://www.mnot.net/blog/2011/07/11/what_proxies_must_do):
```
   A proxy or gateway MUST parse a received Connection header field before
   a message is forwarded and, for each connection-option in this field,
   remove any header field(s) from the message with the same name as the
   connection-option, and then remove the Connection header field itself
   (or replace it with the intermediary's own connection options for the
   forwarded message).
```
So, for request:
```
    GET /foo HTTP/1.1
    Host: example.net
    Connection: Keep-Alive, Foo, Bar
    Foo: abc
    Foo: def
    Keep-Alive: timeout=30
```
The Connection, Keep-Alive and both Foo headers must be removed before forwarding it.
[Co-Advisor HTTP 1.1 compatibility tests](http://coad.measurement-factory.com/cgi-bin/coad2/GraseInfoCgi?info_id=test_clause/rfc2616/hopHdr&proto=http)  follows obsoleted RFC2616, but also checks hop-by-hop headers removing from messages.

'tfw_http_set_hdr_connection()` seems to return without updating Connection header. Also other hop-by hop headers are not removed from test messages.

Used tempesta (`af32a834`) configuration:
```
server <test-server-ip>
cache_methods GET HEAD;
cache_fulfill * *;
```

**Comment by krizhanovsky** [comment]
Please also store correct Server and Via headers in cache as described in #634

**Comment by ikoveshnikov** [comment]
fixed in 72a24ae2cef0cfe19e9e8bc63a716ceb7b2bad4e
